gammapy/maps/geom.py

```diff
@@ -600,6 +600,14 @@ def copy(self):
         """Copy `MapAxis` object"""
         return copy.deepcopy(self)
 
+    def _check_compatibility(self, other):
+        """Check if two axes objects are compatible with each other"""
+        from ..utils.testing import assert_quantity_allclose
+        try:
```

Comment:
This doesn't check e.g. if interpolation mode is compatible.

Suggest to define all comparisons in `__eq__` and then to have this method just be:
```
def check_equal(self, other):
    if self != other:
        raise ValueError('{} not equal'.format(self.__class__.__name__))
```
Ideally there would be a good message about exactly what differs, but that's not easy to achieve, we'd write a ton of code in Gammapy to produce those messages. In principle we could include the `repr` or `str` of the objects in question, and in many cases that would show the difference.